src/test/java/com/google/api/generator/engine/ast/JavaDocCommentTest.java

```diff
@@ -0,0 +1,150 @@
+            .setVariable(Variable.builder().setName("resource").setType(TypeNode.STRING).build())
+            .build();
+    AssignmentExpr assignmentExpr =
+        AssignmentExpr.builder().setVariableExpr(expr).setValueExpr(valueExpr).build();
+    JavaWriterVisitor javaWriterVisitor = new JavaWriterVisitor();
+    assignmentExpr.accept(javaWriterVisitor);
+    String sampleCode = javaWriterVisitor.write();
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode).build();
+    String expected =
+        "sample codes:\n"
+            + "<pre><code>\n"
+            + "resource = project/{project}/shelfId/{shelfId}\n"
+            + "</code></pre>";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createavaDocComment_allComponents() {
+    String content = "this is a test comment";
+    String deprecatedText = "Use the {@link ArchivedBookName} class instead.";
+    String paramName = "shelfName";
+    String paramDescription = "The name of the shelf where books are published to.";
+    String paragraph1 =
+        "This class provides the ability to make remote calls to the backing service through"
+            + " method calls that map to API methods. Sample code to get started:";
+    String paragraph2 =
+        "The surface of this class includes several types of Java methods for each of the API's"
+            + " methods:";
+    List<String> orderedList =
+        Arrays.asList("A flattened method.", "A request object method.", "A callable method.");
+    String throwType = "com.google.api.gax.rpc.ApiException";
+    String throwsDescription = "if the remote call fails.";
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder()
+            .addComment(content)
+            .addParagraph(paragraph1)
+            .addParagraph(paragraph2)
+            .addOrderedList(orderedList)
+            .addParam(paramName, paramDescription)
+            .setThrows(throwType, throwsDescription)
+            .setDeprecated(deprecatedText)
+            .build();
+    String expected =
+        "this is a test comment\n"
+            + "<p> This class provides the ability to make remote calls to the backing service"
+            + " through method calls that map to API methods. Sample code to get started:\n"
+            + "<p> The surface of this class includes several types of Java methods for each of"
+            + " the API's methods:\n"
+            + "<ol>\n"
+            + "<li> A flattened method.\n"
+            + "<li> A request object method.\n"
+            + "<li> A callable method.\n"
+            + "</ol>\n"
+            + "@param shelfName The name of the shelf where books are published to.\n"
+            + "@throws com.google.api.gax.rpc.ApiException if the remote call fails.\n"
+            + "@deprecated Use the {@link ArchivedBookName} class instead.";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+}
```

Comment:
Could we please add a test to exercise only the edge cases of `deprecated` and `throws`?

[Here](https://dzone.com/articles/unit-testing-best-practices-how-to-get-the-most-ou) is a good primer on unit testing best practices.

Reply:
thanks for the resource! It makes sense to cover all the components in unit test. And as mentioned in the last review, technically `setThrows` and `setDeprecated` can be called multiple times, since they are implemented in the same way as addMethod and the getter was removed (added a unit test to show that). I'm not sure they are the `edge cases`  since they have the expected behavior.

Reply:
The expected behavior is that only one `throws` and one `deprecated` will be printed, no matter how many times their respective methods are called, so we need a test case to verify that behavior. The implementation currently does not do that, so let's fix that.

Reply:
We should also test that, no matter what order `setThrows` and `setDeprecated` are called (relative to the other setters and adders), they will always get printed at the end. Same for `addParam`, these should always be printed just before `throws` and `deprecated`. Could you please investigate (e.g. using the [baseline file](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline)) to see if there are  other ordering restrictions we should enforce?